cli/flox/src/commands/init/python.rs

```diff
@@ -0,0 +1,664 @@
+use std::borrow::Cow;
+use std::fmt::{self, Debug, Display};
+use std::path::Path;
+use std::str::FromStr;
+
+use anyhow::{anyhow, Context, Error, Result};
+use flox_rust_sdk::flox::Flox;
+use flox_rust_sdk::models::environment::path_environment::InitCustomization;
+use flox_rust_sdk::models::manifest::PackageToInstall;
+use indoc::{formatdoc, indoc};
+use itertools::Itertools;
+use log::debug;
+
+use super::{format_customization, get_default_package_if_compatible, InitHook, AUTO_SETUP_HINT};
+use crate::commands::init::get_default_package;
+use crate::utils::dialog::{Dialog, Select};
+use crate::utils::message;
+
+pub(super) struct Python {
+    providers: Vec<Provide<Box<dyn Provider>>>,
```

Comment:
note nonblocking: Just as a head up in case we ever want to use the provider pattern more generally or decide it doesn't make sense more generally, I think there are cases where providers need to have knowledge of each other. For Node, for example, if you have a successful nodejs provider but a failing yarn provider, you want to run a nodejs provider without an npm hook. Whereas if you have a successful nodejs provider but no yarn provider at all, you may want to run a nodejs provider with an npm hook.